web-ui/view/gitlab.ml

```diff
@@ -116,12 +122,53 @@ let list_refs ~org ~repo ~refs =
     ]
 
 let list_history ~org ~repo ~ref ~history =
-  Template.instance
-    [
-      breadcrumbs [ (prefix, prefix); (org, org) ] repo;
-      link_gitlab_refs ~org ~repo [ ref ];
-      history_v ~org ~repo ~history;
+  let head_hash =
+    match history with
+    | [] -> ""
+    | (hash, _, _, _)::_ -> hash
+  in
+  let commit_table_div =
+    div
+      ~a:
+        [ a_class [ "bg-gray-50 px-6 py-3 text-gray-500 text-xs font-medium" ] ]
+        (* TODO: We need to start with no stage separation - introduce Analysis/Checks and Build steps later *)
+      [ txt (
+        let len = List.length history in
+        if len = 1 then "Build (1)"
+        else Fmt.str "Builds (%d)" len) ]
+  in
+  let commit_table =
+    List.fold_left
+      (fun l (u, title, status, t) ->
+        let created_at = Timestamps_durations.pp_timestamp (Some t) in
+        List.append l
+          [
+            Build.commit_row ~commit_title:(title) ~short_hash:(short_hash u) ~created_at ~status ~commit_uri:(commit_url ~org ~repo u);
+          ])
```

Comment:
```suggestion
   let f (u, commit_title, status, t) =
        let created_at = Timestamps_durations.pp_timestamp (Some t) in
        let short_hash = short_hash u in
        let commit_uri = commit_url ~org ~repo u in
        Build.commit_row ~commit_title ~short_hash ~created_at ~status ~commit_uri
   in List.map f
```
Isn't it a `List.map` instead of a `List.fold_left`?